### PR TITLE
fix custom key routing

### PR DESCRIPTION
**Pull request by ahmedbally**
support laravel 8 custom key routing
https://laravel.com/docs/8.x/routing#customizing-the-default-key-name

### CODE REVIEW

File: src/Traits/OptimusEncodedRouteKey.php [hunk 1]
```diff
@@ -29,13 +29,11 @@ public function resolveRouteBinding($value, $field = null)
...
-        if (!is_int($value)) {
-            return null;
+        if (is_int($value) && is_null($field)){
+            $value = $this->getOptimus()->decode($value);
         }
 
-        $id = $this->getOptimus()->decode($value);
-
-        return $this->where($this->getRouteKeyName(), '=', $id)->first();
+        return $this->where($field ?? $this->getRouteKeyName(),$value)->first();
```
Comment (antonkomarev) [inline src/Traits/OptimusEncodedRouteKey.php:36]: space after `,` is missing

**Comment by antonkomarev** [comment]
Thank you for the contribution @ahmedbally! It will be great to cover this in tests to not break this feature in the future: https://github.com/cybercog/laravel-optimus/blob/master/tests/Traits/OptimusEncodedRouteKeyTest.php

**Comment by ahmedbally** [comment]
I hope you find it well
I tried my best 👍 
if you could re-run tests it will  give you success

**Comment by antonkomarev** [comment]
Your code is OK, it's scrutenizer config issue. Thank you for your efforts. I will test it and release then.

**Comment by antonkomarev** [comment]
I've tested it, this way we have a breaking change. For example when we have our encoded key stored in `public_id` property. Then we can not bypass it anymore.

This would not work anymore:
```php
Route::get('/posts/{public_id:encodedId}', function (Post $post) {
    return $post;
});
```

**Comment by antonkomarev** [comment]
But since originally we haven't used `$field` parameter - it is not a case. Nobody can use it this way.

**Comment by antonkomarev** [comment]
Merged it, and will do small tweak:
```php
public function resolveRouteBinding($value, $field = null)
{
    if ($field === null) {
        $field = $this->getRouteKeyName();
    }

    if (is_string($value) && ctype_digit($value)) {
        $value = (int) $value;
    }

    if (is_int($value) && $field === $this->getRouteKeyName()) {
        $value = $this->getOptimus()->decode($value);
    }

    return $this->where($field, $value)->first();
}
```

It will allow us to explicitly use route key too:
```php
Route::get('/posts/{id:encodedId}', function (Post $post) {
    return $post;
});
```

**Comment by ahmedbally** [comment]
this get into my mind when i  was taking a shower 😂 😂 😂

**Comment by ahmedbally** [comment]
thank you for your tweaks

**Comment by antonkomarev** [comment]
[Laravel Optimus v3.7.0](https://github.com/cybercog/laravel-optimus/releases/tag/3.7.0) has been released! @ahmedbally you are in contributors list now :1st_place_medal:

**Comment by antonkomarev** [comment]
If you have a twitter - I could mention you in a release post.

**Comment by ahmedbally** [comment]
I am very happy to be in contributors list
that's my twitter 
https://twitter.com/ahmedbally
thanks

**Pull request merged**